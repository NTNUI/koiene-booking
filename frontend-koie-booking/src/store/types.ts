// rootState = generic type, defines the root state type
import AdminBooking, { AdminBookingDictionary } from '@/types/admin/AdminBooking';
import { AdminReportKoieDictionary } from '@/types/admin/AdminReports';

export interface RootState {
  version: string;
  koie: KoieState;
  booking: BookingState;
  auth: AuthState;
  avalanche: AvalancheState;
  adminBookings: AdminBookingsState;
}

export interface KoieState {
  currentLocale: string;
  isLoading: boolean;
  koieData: Koie;
  error: boolean;
  allKoier: [];
}

export interface Koie {
  location: {
    latitude: string;
    longitude: string;
    area: string;
    difficultyInfo_eng: string;
    difficultyInfo_nor: string;
    altitude: number;
    terrain_eng: string;
    terrain_nor: string;
    map_pdf: string;
    kartblad: string;
  };
  description: {
    yr_link_eng: string;
    yr_link_nor: string;
    description_eng: string;
    description_nor: string;
    directions_eng: string;
    directions_nor: string;
    parking_eng: string;
    parking_nor: string;
  };
  beds_available_in_booking_window: Array<BedsAvailable>;
  name: string;
  number_of_beds: number;
  booking_window: number;
  price_member: number;
  price_not_member: number;
}

export interface BedsAvailable {
  date: string;
  bedsAvailable: number;
}

export interface BookingState {
  isLoading: boolean;
  dateFrom: string;
  dateTo: string;
  beds: number;
  step: number;
  guests: Array<Guest>;
  numberOfMembers: number;
  numberOfNonMembers: number;
  edited: boolean;
  validForm: boolean;
  unavailableDates: [];
  availableBeds: number;
  noDatesAvailable: boolean;
  error: boolean;
  bookingData: BookingInfo;
  paymentConfirmed: boolean;
}

export interface ReportState {
  validForm: boolean;
  edited: boolean;
  isLoading: boolean;
  error: boolean;
  step: number;
  reportData: ReportData;
  bookingInfo: BookingInfo;
}

export interface ReportData {
  id: number;
  booking_uuid: string;
  firewood: number;
  chopped_up_wood: number;
  smoke_detector_is_working: boolean;
  gas_is_full: boolean;
  gas_burner_primus: number;
  axe: number;
  hammer: number;
  saw: number;
  saw_blade: number;
  saw_bench: number;
  spade: number;
  kerosene_lamp: number;
  detergent: number;
  dishware: number;
  cookware: number;
  cabin_book: number;
  candle_holders: number;
  fire_blanket: number;
  fire_extinguisher: number;
  other_faults: string;
  boat_status: number;
  canoe_status: number;
  life_jackets_status: number;
  feedback: string;
}

export interface AvalancheState {
  warningLevels: Array<Warning>;
  disableBooking: boolean;
  warningData: Array<WarningResult>;
  koierToCheck: Array<string>;
}

export interface Warning {
  day: string;
  level: number;
}

export interface WarningResult {
  dangerLevel: number;
  validTo: string;
}

export interface CreateBookingInfo {
  koie: string;
  from_date: string;
  to_date: string;
  guests: Array<Guest>;
  guests_member: number;
  guests_not_member: number;
}

export interface BookingInfo {
  id: number;
  uuid: string;
  user: number;
  koie: string;
  price: number;
  booking_transaction_id: string;
  from_date: string;
  to_date: string;
  guests_member: number;
  guests_not_member: number;
}

export interface AuthState {
  isAdmin: boolean;
  tokens: {
    access: string | null;
    refresh: string | null;
  };
}

export interface Guest {
  name: string;
  number: string;
  email: string;
  isMember: boolean;
}

export interface AdminBookingsState {
  cabinsWithBookings: AdminBookingDictionary;
  startDate: string;
}

export interface AdminReportsState {
  chosenCabin: string;
  expanded: boolean;
  cabins: AdminReportKoieDictionary;
}
