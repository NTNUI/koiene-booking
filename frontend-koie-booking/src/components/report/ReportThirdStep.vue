<template>
  <v-layout :class="$style.container" :dark="true">
    <h1 :class="$style.heading">{{ $t('report.step3') }}</h1>
    <v-form v-model="validForm" :class="$style.form">
      <v-layout :class="$style.separator">
        <h3 class="py-4" :class="$style.supplyRegistration">{{ $t('report.water_equipment.boat') }}</h3>
        <v-layout class="px-4">
          <v-radio-group v-model="boatStatus" mandatory required :color="$scssVars.globalColorBackgroundLight">
            <v-radio :value="3" :label="$t('report.water_equipment.missing_boat')"></v-radio>
            <v-radio :value="1" :label="$t('report.water_equipment.broken')"></v-radio>
            <v-radio :value="2" :label="$t('report.water_equipment.missing_paddles')"></v-radio>
            <v-radio :value="4" :label="$t('report.water_equipment.unsure')"></v-radio>
            <v-radio :value="0" :label="$t('report.water_equipment.no_faults')"></v-radio>
          </v-radio-group>
        </v-layout>
      </v-layout>
      <v-layout :class="$style.separator">
        <h3 class="py-4" :class="$style.supplyRegistration">{{ $t('report.water_equipment.canoe') }}</h3>
        <v-layout class="px-4">
          <v-radio-group v-model="canoeStatus" mandatory required :color="$scssVars.globalColorBackgroundLight">
            <v-radio :value="3" :label="$t('report.water_equipment.missing_canoe')"></v-radio>
            <v-radio :value="1" :label="$t('report.water_equipment.broken')"></v-radio>
            <v-radio :value="2" :label="$t('report.water_equipment.missing_paddles')"></v-radio>
            <v-radio :value="4" :label="$t('report.water_equipment.unsure')"></v-radio>
            <v-radio :value="0" :label="$t('report.water_equipment.no_faults')"></v-radio>
          </v-radio-group>
        </v-layout>
      </v-layout>
      <v-layout :class="$style.separator">
        <h3 class="py-4" :class="$style.supplyRegistration">{{ $t('report.water_equipment.life_jackets') }}</h3>
        <v-layout class="px-4">
          <v-radio-group v-model="lifeJacketsStatus" mandatory required :color="$scssVars.globalColorBackgroundLight">
            <v-radio :value="2" :label="$t('report.water_equipment.missing_life_jackets')"></v-radio>
            <v-radio :value="1" :label="$t('report.water_equipment.broken')"></v-radio>
            <v-radio :value="3" :label="$t('report.water_equipment.unsure')"></v-radio>
            <v-radio :value="0" :label="$t('report.water_equipment.no_faults')"></v-radio>
          </v-radio-group>
        </v-layout>
      </v-layout>
    </v-form>
  </v-layout>
</template>

<script lang="ts">
import Vue from 'vue';
import { ReportThirdStepData } from '../../types/report';

export default Vue.extend({
  name: 'ReportThirdStep',
  data(): ReportThirdStepData {
    return {
      validForm: true,
    };
  },
  computed: {
    boatStatus: {
      get() {
        return this.$store.state.report.reportData.boat_status;
      },
      set(value: number) {
        this.$store.commit('report/setBoatStatus', value);
        this.$store.commit('report/setEdited', true);
      },
    },
    canoeStatus: {
      get() {
        return this.$store.state.report.reportData.canoe_status;
      },
      set(value: number) {
        this.$store.commit('report/setCanoeStatus', value);
        this.$store.commit('report/setEdited', true);
      },
    },
    lifeJacketsStatus: {
      get() {
        return this.$store.state.report.reportData.life_jackets_status;
      },
      set(value: number) {
        this.$store.commit('report/setLifeJacketsStatus', value);
        this.$store.commit('report/setEdited', true);
      },
    },
  },
});
</script>

<style lang="scss" module>
.container {
  display: flex;
  flex-direction: column;
}
.heading {
  margin-top: 20px;
  align-self: center;
}
.separator {
  border-left: solid 4px;
  border-color: #2f3a50;
  display: flex;
  flex-direction: column;
  margin-top: 24px;
}
.separator > h3 {
  padding: 16px;
}
.feedbackField {
  width: 100%;
  padding-right: 10px;
}
</style>
