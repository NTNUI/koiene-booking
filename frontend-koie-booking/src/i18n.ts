import Vue from 'vue';
import VueI18n, { LocaleMessages } from 'vue-i18n';
import enJson from './locales/en.json';

Vue.use(VueI18n);

function loadLocaleMessages(): LocaleMessages {
  // Testing currently only supports manually importing of json-files. English is the only language that gets loaded now
  let locales: any;
  const messages: LocaleMessages = {};
  if (process.env.NODE_ENV === 'test') {
    messages['en'] = enJson;
  } else {
    locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
    locales.keys().forEach((key: any) => {
      const matched = key.match(/([A-Za-z0-9-_]+)\./i);
      if (matched && matched.length > 1) {
        const locale = matched[1];
        messages[locale] = locales(key);
      }
    });
  }
  return messages;
}

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || 'en',
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || 'en',
  messages: loadLocaleMessages()
});
