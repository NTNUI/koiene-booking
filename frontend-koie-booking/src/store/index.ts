import Vue from 'vue';
import Vuex, { StoreOptions } from 'vuex';
import axios from 'axios';
import VueAxios from 'vue-axios';
import { RootState } from './types';
import { koie } from './modules/users/koie/index';
import { booking } from './modules/users/booking/index';
import { avalanche } from './modules/users/avalanche/index';
import { report } from './modules/users/report/index';
import { auth } from './modules/users/auth/index';
import { adminBookings } from '@/store/modules/admin/bookings';

Vue.use(Vuex, VueAxios, axios);

const store: StoreOptions<RootState> = {
  modules: {
    adminBookings,
    koie,
    booking,
    avalanche,
    report,
    auth
  }
};

export default new Vuex.Store<RootState>(store);

export { store as storeConfig };
