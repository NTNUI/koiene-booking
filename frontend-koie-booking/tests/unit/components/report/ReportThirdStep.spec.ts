import Vue from 'vue';
import Vuetify from 'vuetify';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import { storeConfig } from '@/store';
import { cloneDeep } from 'lodash';
import i18n from '@/i18n';
import scssVars from '@/styles/variables.scss';

Vue.use(Vuetify);

// Utilities
import { mount, createLocalVue } from '@vue/test-utils';

// Components or views
import ReportThirdStep from '@/components/report/ReportThirdStep.vue';

describe('Component ReportThirdStep.vue', () => {
  let wrapper: any;
  let localVue: any;
  let vuetify: any;
  let store: any;

  beforeEach(() => {
    localVue = createLocalVue();
    vuetify = new Vuetify();
    localVue.use(Vuetify);
    localVue.use(Vuex);
    store = new Vuex.Store(cloneDeep(storeConfig));
    localVue.prototype.$scssVars = scssVars;
  });

  it('Matches snapshot', async () => {
    wrapper = mount(ReportThirdStep, {
      localVue,
      vuetify,
      i18n,
      store,
    });

    expect(wrapper).toMatchSnapshot();
  });
});
