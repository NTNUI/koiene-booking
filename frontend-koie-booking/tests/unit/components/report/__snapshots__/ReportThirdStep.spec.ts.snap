// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ReportThirdStep.vue Matches snapshot 1`] = `
<div class="layout dark container">
  <h1 class="heading">Equipment</h1>
  <form novalidate="novalidate" class="v-form">
    <div class="layout separator">
      <h3 class="py-4">What was the status for the boat?</h3>
      <div class="layout px-4">
        <div class="v-input theme--light v-input--selection-controls v-input--radio-group v-input--radio-group--column">
          <div class="v-input__control">
            <div class="v-input__slot" style="height: auto;">
              <div role="radiogroup" aria-labelledby="input-9" class="v-input--radio-group__input">
                <div class="v-radio theme--light">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-10" role="radio" type="radio" name="radio-9" value="3">
                    <div class="v-input--selection-controls__ripple primary--text"></div>
                  </div><label for="input-10" class="v-label theme--light" style="left: 0px; position: relative;">Missing boat</label>
                </div>
                <div class="v-radio theme--light">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-12" role="radio" type="radio" name="radio-9" value="1">
                    <div class="v-input--selection-controls__ripple primary--text"></div>
                  </div><label for="input-12" class="v-label theme--light" style="left: 0px; position: relative;">Broken / useless</label>
                </div>
                <div class="v-radio theme--light">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-14" role="radio" type="radio" name="radio-9" value="2">
                    <div class="v-input--selection-controls__ripple primary--text"></div>
                  </div><label for="input-14" class="v-label theme--light" style="left: 0px; position: relative;">Missing paddles</label>
                </div>
                <div class="v-radio theme--light">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-16" role="radio" type="radio" name="radio-9" value="4">
                    <div class="v-input--selection-controls__ripple primary--text"></div>
                  </div><label for="input-16" class="v-label theme--light" style="left: 0px; position: relative;">Didn't use / don't know</label>
                </div>
                <div class="v-radio theme--light v-item--active">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-marked theme--light primary--text"></i><input aria-checked="true" id="input-18" role="radio" type="radio" name="radio-9" value="0">
                    <div class="v-input--selection-controls__ripple primary--text"></div>
                  </div><label for="input-18" class="v-label theme--light" style="left: 0px; position: relative;">No faults</label>
                </div>
              </div>
            </div>
            <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
          </div>
        </div>
      </div>
    </div>
    <div class="layout separator">
      <h3 class="py-4">What was the status for the canoe?</h3>
      <div class="layout px-4">
        <div class="v-input theme--light v-input--selection-controls v-input--radio-group v-input--radio-group--column">
          <div class="v-input__control">
            <div class="v-input__slot" style="height: auto;">
              <div role="radiogroup" aria-labelledby="input-22" class="v-input--radio-group__input">
                <div class="v-radio theme--light">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-23" role="radio" type="radio" name="radio-22" value="3">
                    <div class="v-input--selection-controls__ripple primary--text"></div>
                  </div><label for="input-23" class="v-label theme--light" style="left: 0px; position: relative;">Missing canoe</label>
                </div>
                <div class="v-radio theme--light">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-25" role="radio" type="radio" name="radio-22" value="1">
                    <div class="v-input--selection-controls__ripple primary--text"></div>
                  </div><label for="input-25" class="v-label theme--light" style="left: 0px; position: relative;">Broken / useless</label>
                </div>
                <div class="v-radio theme--light">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-27" role="radio" type="radio" name="radio-22" value="2">
                    <div class="v-input--selection-controls__ripple primary--text"></div>
                  </div><label for="input-27" class="v-label theme--light" style="left: 0px; position: relative;">Missing paddles</label>
                </div>
                <div class="v-radio theme--light">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-29" role="radio" type="radio" name="radio-22" value="4">
                    <div class="v-input--selection-controls__ripple primary--text"></div>
                  </div><label for="input-29" class="v-label theme--light" style="left: 0px; position: relative;">Didn't use / don't know</label>
                </div>
                <div class="v-radio theme--light v-item--active">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-marked theme--light primary--text"></i><input aria-checked="true" id="input-31" role="radio" type="radio" name="radio-22" value="0">
                    <div class="v-input--selection-controls__ripple primary--text"></div>
                  </div><label for="input-31" class="v-label theme--light" style="left: 0px; position: relative;">No faults</label>
                </div>
              </div>
            </div>
            <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
          </div>
        </div>
      </div>
    </div>
    <div class="layout separator">
      <h3 class="py-4">What was the status for the life jackets?</h3>
      <div class="layout px-4">
        <div class="v-input theme--light v-input--selection-controls v-input--radio-group v-input--radio-group--column">
          <div class="v-input__control">
            <div class="v-input__slot" style="height: auto;">
              <div role="radiogroup" aria-labelledby="input-35" class="v-input--radio-group__input">
                <div class="v-radio theme--light">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-36" role="radio" type="radio" name="radio-35" value="2">
                    <div class="v-input--selection-controls__ripple primary--text"></div>
                  </div><label for="input-36" class="v-label theme--light" style="left: 0px; position: relative;">Missing / too few life jackets</label>
                </div>
                <div class="v-radio theme--light">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-38" role="radio" type="radio" name="radio-35" value="1">
                    <div class="v-input--selection-controls__ripple primary--text"></div>
                  </div><label for="input-38" class="v-label theme--light" style="left: 0px; position: relative;">Broken / useless</label>
                </div>
                <div class="v-radio theme--light">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-40" role="radio" type="radio" name="radio-35" value="3">
                    <div class="v-input--selection-controls__ripple primary--text"></div>
                  </div><label for="input-40" class="v-label theme--light" style="left: 0px; position: relative;">Didn't use / don't know</label>
                </div>
                <div class="v-radio theme--light v-item--active">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-marked theme--light primary--text"></i><input aria-checked="true" id="input-42" role="radio" type="radio" name="radio-35" value="0">
                    <div class="v-input--selection-controls__ripple primary--text"></div>
                  </div><label for="input-42" class="v-label theme--light" style="left: 0px; position: relative;">No faults</label>
                </div>
              </div>
            </div>
            <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
